calc.rb

```diff
@@ -0,0 +1,76 @@
+puts "The Calculator"
+puts "I'll take two numbers and perform a mathematical operation on them. Let's try."
+
+# First number
+puts "\nWhat is the first number?"
+num1 = gets.chomp
+
+# Handling invalid user input for first number
+if num1 == nil || (num1.to_i.to_s != num1 && num1.to_f.to_s != num1)
+	puts "Please type a valid first number. For example, type 3 instead of three."
+	num1 = gets.chomp
+end
+
+# Second number
+puts "What is the second number?"
+num2 = gets.chomp
+
+# Handling invalid user input for second number
+if num2 == nil || (num2.to_i.to_s != num2 && num2.to_f.to_s != num2)
+	puts "Please type a valid second number. For example, type 5 instead of five."
+	num2 = gets.chomp
+end
+
+# Convert number to integer or float
+if num1.include? "."
+	num1 = num1.to_f
+else
+	num1 = num1.to_i
+end
```

Comment:
I like this approach of determining if we need a float or integer. It's very clear about what it's testing and how it's changing the input.